### PR TITLE
Bugfix/issue 95 worker mem leak 2

### PR DESCRIPTION
**Pull request by stephen-palmer**
- Remove the ClusterManager module, which was not providing much utility. Replaced usage with direct `cluster` module implementation in the ReliabilityManager
- Removed ref counting from CacheFS which was used to prevent files from changing while they were being served. This was in practice an uncommon edge case to worry about, and not worth the additional memory and performance overhead (especially on very large projects).
- The test for preventing file overwrites was moved out of the common API suite and into to the CacheRAM specific test suite, since CacheFS would fail this test now with ref counting removed.
- A few minor syntax and test fixes discovered along the way

### CODE REVIEW

File: lib/cache/reliability_manager.js [hunk 1]
```diff
@@ -30,9 +31,13 @@ class ReliabilityManager {
             }
         }
 
-        cm.listenFor("_updateReliabilityFactorForVersion", (data) => {
-            return this._updateReliabilityFactorForVersion(data);
-        });
+        if(cluster.isMaster) {
+            cluster.on('message', (worker, msg) => {
+                if(msg._msg && msg._msg === this._kMsgUpdateReliabilityFactor) {
```
[the remark this answers is not in the document]
Comment (stephen-palmer) [reply lib/cache/reliability_manager.js]: good catch. updated.
